inc/functions.php

```diff
@@ -549,6 +549,64 @@ function my_date($format, $stamp=0, $offset="", $ty=1, $adodb=false)
 	return $date;
 }
 
+/**
+ * Get a mail handler instance, a MyBB's built-in SMTP / PHP mail hander or one created by a plugin.
+ * @param bool $use_buitlin Whether to use MyBB's built-in mail handler.
+ *
+ * @return object A MyBB built-in mail handler or one created by plugin(s).
+ */
+function &get_my_mailhandler($use_buitlin = false)
+{
+	global $mybb, $plugins;
+	static $my_mailhandler;
+	static $my_mailhandler_builtin;
```

Comment:
Seems like in a default install both `$my_mailhandler_builtin` and `$my_mailhandler` would store the same data, is there a real need to keep both variables?

Reply:
Perhaps a custom mail handler is a wrapper of builtin mail handler? So when creating a custom one, it could get the default builtin one first.

Reply:
And by the way, if a custom mail handler is not created, `$my_mailhandler` contains a reference to the object stored in `$my_mailhandler_builtin`.

Reply:
I will concede on this one then.